From 3cdf3caadf475681bf87869e8056124f98a6c3ad Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Mon, 30 May 2016 03:45:25 +0100
Subject: [PATCH 07/13] xterm-no-tigetstr-on-tty

user:        Fabian Groffen <grobian@gentoo.org>
date:        Sun Oct 27 16:31:36 2013 +0100
files:       curs_main.c
description:
xterm: check we're on a tty before calling tigetstr

tigetstr appears to spinlock/hang when called without a tty, which is
undesirable when mutt is ran from cron or ssh in a scripted fashion,
e.g. to send mail.  Hence, abort early if stdout doesn't appear to be
connected to a tty.
Fixes https://bugs.gentoo.org/show_bug.cgi?id=432562

---
 curs_main.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/curs_main.c b/curs_main.c
index 9a2bf2a..3021ecc 100644
--- a/curs_main.c
+++ b/curs_main.c
@@ -154,6 +154,11 @@ short mutt_ts_capability(void)
     NULL
   };
 
+  /* check that we're actually having a tty, such that tigetstr doesn't
+   * hang/spinlock */
+  if (!isatty(1))
+    return 0;
+
   /* If tsl is set, then terminfo says that status lines work. */
   tcaps = tigetstr("tsl");
   if (tcaps && tcaps != (char *)-1 && *tcaps)
-- 
2.8.2

